### PR TITLE
refactors kubeapiserver webhook utility to be generic

### PR DESCRIPTION
**Pull request by pbarker**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
The dynamic audit work https://github.com/kubernetes/enhancements/issues/600 would like to reuse this utility for outgoing webhooks that is currently specific to admission.

**Does this PR introduce a user-facing change?**:
```
NONE
```

### CODE REVIEW

File: pkg/kubeapiserver/admission/config.go
Comment (pbarker) [reply pkg/kubeapiserver/admission/config.go]: not really, I'll change it to a named param

File: pkg/kubeapiserver/util/webhook.go
Comment (yue9944882) [inline pkg/kubeapiserver/util/webhook.go]: @pbarker i guess u want to move it to `staging/apiserver` unless u r sure i wont be reused from staging area?

Comment (pbarker) [reply pkg/kubeapiserver/util/webhook.go]: good idea, updating

File: staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go
Comment (yue9944882) [inline staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go]: the hard-coding looks odd tho 😥but it should be another issue

**Review by yue9944882 — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go
Comment (deads2k) [reply staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go]: > the hard-coding looks odd tho but it should be another issue

The hardcoding is based on the idea that the kube apiserver (in that hardcoded location in a cluster), can be trusted with a token intended for the API.  Since SA tokens are auto-mounted and intended for the API, that host is safe to trust.

Comment (yue9944882) [reply staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go]: then we shouldnt move these hardcoding (L44 also) into generic apiserver, amright? since it only applies to kube apiserver.

File: staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go
Comment (yue9944882) [reply staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go]: > kubeapiserverClientConfig

makes sense

**Comment by pbarker** [comment]
@yue9944882 could I get your lgtm one more time for the param name change?

**Comment by yue9944882** [comment]
/hold

oh sry. @pbarker  plz change that param name

**Pull request merged**
